### ISSUE TITLE
Scp not working even on standard port

### ISSUE THREAD

**Issue by PurityControl** [blinksh/blink#624]
I received an update today saying scp was fixed for non standard ports. However, I have been unable to get scp working on any port.

`scp iCloud/myfile.txt ubuntuvm:~`

Gives the following error:

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:05 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:05 --:--:--     0
curl: (51) SSL peer certificate or SSH remote key was not OK
```
This error is the same if I replace ubuntuvm with its IP address.
The ssh remote key is fine as I can `ssh ubuntuvm` without issue in the same shell.

**Comment by camfowler** [comment]
I get the same error when trying to transfer a file in either direction.

**Comment by camfowler** [comment]
From the linked issue, this comment explains the cause of the issue, https://github.com/blinksh/blink/issues/564#issuecomment-438658248

**Comment by casperghst42** [comment]
Now almost a year later, this is still not working. It would be nice if this is moved to the stop of list of things to fix!!!

**Comment by yury** [comment]
Yep, this is top priority. But requires to move to OpenSSH. This is a ton of work, that we will start after v13 release.

**Comment by casperghst42** [comment]
> Yep, this is top priority. But requires to move to OpenSSH. This is a ton of work, that we will start after v13 release.

I and many others have paid for this app, which has a major problem ... I do understand that it takes time to fix issues, but is a major flaw which should be listed as a “know issue” on the AppStore.

**Comment by yury** [comment]
It actually works. If you connect to host with ssh2 command first. scp is implemented with libcurl, which is not understand any other keys except RSA. Sorry for that.

**Comment by casperghst42** [comment]
> It actually works. If you connect to host with ssh2 command first. scp is implemented with libcurl, which is not understand any other keys except RSA. Sorry for that.

1) ‘ssh2’ to the destination server
2) ‘scp <source filename> <user>@<dest server>:~/<dest filename>‘

That works... but it would be nice if one does not have to google for this.

**Comment by yury** [comment]
We have few options here.
1. Fix `scp` with porting to OpenSSH
2. Put notice in the help
3. Warn when scp fails

**Comment by ddo** [comment]
how about scp with password? not public key?

**Comment by demyxco** [comment]
Hello @yury as you mentioned that scp is fixed but I still can’t copy files from remote to my device. Is ed25519 still not supported?

**Comment by Vinznx** [comment]
I can ssh to the server. But I cannot use scp to transport files. I try to use ssh2 first. But scp still does not work.

**Comment by j3r1ch0** [comment]
Two years later and this is still not working! I and many other paid $20 for this app, and this doesn’t even work! There’s no warning, no comment on the App Store page, nothing. It’s not even like this is impossible to fix. Half the reason I bought this app was so that I could easily copy things from the terminal on my remote servers to my iPhone and iPad. That’s impossible with the ssh2 workaround. So what’s the deal with this? Will it ever be fixed or should I start asking for a refund? Two years is plenty of time where this should’ve been handled.

**Comment by yury** [comment]
@j3r1ch0 we are working on new scp implementation. Hope to push it soon

**Comment by kantgreven** [comment]
Any news around when this will be fixed? I’m still having trouble using scp as of Blink 13.5.10. I’ve tried with the ssh2 solution, specifying the path with no default home folder and even passing the — insecure argument. No luck, still curl(67) error, really frustrating. Will this be resolved in Blink version 14? And are there any news on when version 14 will be released?

**Comment by kantgreven** [comment]
Okay, it seems I found a somewhat not-pretty solution for my problem.

1. Make sure you select your key when creating a host via config
2. ssh2 upon first connection
3. ssh-copy-id [your key] [your host]

Works after that

**Comment by yury** [comment]
scp is totally rewritten now. This is working now.
